site/src/pages/docs/client-factory-builder.mdx

```diff
@@ -0,0 +1,140 @@
+# Customizing a `ClientFactory` with `ClientFactoryBuilder`
+
+A <type://ClientFactoryBuilder> is a builder class that enables you to create a <type://ClientFactory>
+with custom options. For example, you might need to use the `connectTimeoutMillis` method to adjust the
+connection timeout of your clients.
+
+```java
+try (ClientFactory factory =
+          ClientFactory.builder()
+                       // Set the connection timeout to 5 seconds.
+                       .connectTimeoutMillis(5000)
+                       .build()) {
+   final WebClient client = WebClient.builder()
+                                     .factory(factory)
+                                     .build();
+   // do something with web client
+  }
+```
+
+As shown in the above example, you can specify a custom <type://ClientFactory> when building a client
+via the <type://ClientFactoryBuilder#factory()> method.
+The client created with the given <type://ClientFactory> will respect the settings such as connection timeout
+as you defined.
+
+## What options are available?
+
+See <type://ClientFactoryBuilder> for the complete list of customizable options.
+
+## Setting options with `ClientOption`
```

Comment:
How about removing this section and just introducing `ClientFactoryOptions`?
```
You might want to create another <type://ClientFactory> that has most of the same properties from the <type://ClientFactory> you just built, but only a few are different. In that case,
you can use <type://ClientFactoryOptions>
```

Reply:
I think describing `ClientFactoryOptions` in previous section is good idea (it is too verbose), let me utilize/re-use existing paragraph.